app/actions/remote/groups.ts

```diff
@@ -54,15 +54,15 @@ export const fetchGroupsByNames = async (serverUrl: string, names: string[], fet
 
 export const fetchGroupsForChannel = async (serverUrl: string, channelId: string, fetchOnly = false) => {
     try {
-        const operator = getOperator(serverUrl);
+        const {operator} = DatabaseManager.getServerDatabaseAndOperator(serverUrl);
         const client = NetworkManager.getClient(serverUrl);
         const response = await client.getAllGroupsAssociatedToChannel(channelId);
 
         if (!fetchOnly) {
-            return await storeGroups(serverUrl, response.groups);
+            return storeGroups(serverUrl, response.groups);
         }
 
-        return await prepareGroups(operator, response.groups);
+        return prepareGroups(operator, response.groups);
     } catch (error) {
         forceLogoutIfNecessary(serverUrl, error as ClientErrorProps);
```

Comment:
0/5 due to the database manager being able to throw, there is the possibility this is not a ClientError, so not sure if this will behave nicely.

Reply:
I think it only logs you out if both the database and current user checks have failed, so I think it's fine.

Reply:
This works just fine in the case of any error :)